apps/src/codebridge/Console/ConsoleManager.ts

```diff
@@ -78,6 +80,20 @@ export default class ConsoleManager {
     this.inputBuffer = '';
   }
 
+  public addTerminalLinesListener(listener: () => void) {
+    this.terminalLinesListeners.push(listener);
+  }
+
+  public removeTerminalLinesListener(listener: () => void) {
+    this.terminalLinesListeners = this.terminalLinesListeners.filter(
+      l => l !== listener
+    );
+  }
+
+  private executeTerminalLinesListeners() {
+    this.terminalLinesListeners.forEach(listener => listener());
```

Comment:
Could we simplify the logic here by having the listener take a parameter of `terminalLines`, so we don't have to call back to get the terminal lines when the listener executes?

Reply:
Great idea - updated!